edx_repo_tools/oep2/explode_repos_yaml.py

```diff
@@ -48,62 +57,181 @@ def explode(hub, dry):
+                    )
+        else:
+            dry_echo(
+                dry,
+                textwrap.dedent("""\
+                    Created pull request {repo}#{number}: {title}
+                        URL: {url}
+                """).format(
+                    url=pull.html_url if not dry else "N/A",
+                    repo=gh_repo.full_name,
+                    number=pull.number if not dry else "XXXX",
+                    title=pull.title if not dry else pr_title,
+                ),
+                fg='green'
+            )
 
-        gh_repo.create_ref(
-            'refs/heads/{}'.format(BRANCH_NAME),
-            gh_repo.branch('master').commit.sha,
-        )
-        gh_repo.create_file(
-            path=OPEN_EDX_YAML,
-            message='Add an OEP-2 compliant openedx.yaml file',
-            content=file_contents,
-            branch=BRANCH_NAME,
-        )
-        pull = gh_repo.create_pull(
-            title='Add an OEP-2 compliant openedx.yaml file',
-            base='master',
-            head=BRANCH_NAME,
-        )
-        click.secho('Created pull request {} against {}'.format(
-            pull.html_url,
-            repo,
-        ), fg='green')
+            if not dry:
+                pull = gh_repo.create_pull(
+                    title=pr_title,
+                    base=gh_repo.default_branch,
+                    head=BRANCH_NAME,
+                    body=pr_body
+                )
 
 
 @click.command()
 @pass_github
 @click.option('--org', multiple=True, default=['edx', 'edx-ops'])
-def implode(hub, org):
+@click.option(
+    '--branch',
+    multiple=True,
+    default=None,
+    help="Which branches to look at for openedx.yaml contents. Earlier branches will get priority"
+)
+def implode(hub, org, branch):
     """
     Implode all openedx.yaml files, and print the results as formatted output.
     """
-    data = dict(iter_openedx_yaml(hub, org))
+    data = dict(iter_openedx_yaml(hub, org, branch))
     click.echo(yaml.safe_dump(data, encoding=None, indent=4))
 
 
-def iter_openedx_yaml(hub, orgs):
+def iter_openedx_yaml(hub, orgs, branches=None):
+    if branches is None:
+        branches = []
+
     for org in orgs:
         for repo in hub.organization(org).iter_repos():
             if repo.fork:
                 LOGGER.debug("Skipping %s because it is a fork", repo.full_name)
                 continue
 
-            contents = repo.contents(OPEN_EDX_YAML)
-            if contents is None:
-                LOGGER.debug("Skipping %s because there is no %s", repo.full_name, OPEN_EDX_YAML)
-                continue
+            for branch in branches:
+                contents = repo.contents(OPEN_EDX_YAML, ref=branch)
+                if contents is not None:
+                    LOGGER.debug("Found openedx.yaml at %s:%s", repo.full_name, branch)
+                    yield repo.full_name, yaml.safe_load(contents.decoded)
+                    break
+            else:
```

Comment:
Seems like this else would be better handled by appending repo.default_branch to branches, and then letting the loop handle it.